src/components/CopyAdForm/CopyAdForm.tsx

```diff
@@ -0,0 +1,281 @@
+import { Controller, FormProvider, NonUndefined, useForm } from 'react-hook-form';
+import { TCountryListItem, TCurrency, THooks } from 'types';
+import { RATE_TYPE } from '@/constants';
+import { api, useFloatingRate } from '@/hooks';
+import { AdFormInput } from '@/pages/my-ads/components/AdFormInput';
+import { formatTime, getValidationRules, restrictDecimalPlace } from '@/utils';
+import { Localize, useTranslations } from '@deriv-com/translations';
+import { InlineMessage, Text, useDevice } from '@deriv-com/ui';
+import { FloatingRate } from '../FloatingRate';
+import CopyAdFormDisplayWrapper from './CopyAdFormDisplayWrapper';
+import './CopyAdForm.scss';
+
+type TSavedFormValues = {
+    amount: number;
+    maxOrder: string;
+    minOrder: string;
+    rateValue: string;
+};
+
+type TCopyAdFormProps = NonUndefined<THooks.AdvertiserAdverts.Get>[0] & {
+    formValues: TSavedFormValues;
+} & {
+    isModalOpen: boolean;
+    onClickCancel: (values: TSavedFormValues) => void;
+    onFormSubmit: (values: TSavedFormValues) => void;
+    onRequestClose: () => void;
+};
+
+type TFormValues = {
+    amount: number;
+    'float-rate-offset-limit': string;
+    'max-order': string;
+    'min-order': string;
+    'rate-type-string': string;
+    'rate-value': string;
+};
+
+const CopyAdForm = ({
+    formValues,
+    isModalOpen,
+    onClickCancel,
```

Comment:
can't we use the `onRequestClose` for this? 🤔

Reply:
i have removed the prop. its not needed.